app/src/main/java/org/jellyfin/androidtv/details/DetailsActivity.kt

```diff
@@ -33,14 +29,13 @@ class DetailsActivity : FragmentActivity() {
 
 		GlobalScope.launch(Dispatchers.Main) {
 			val baseItem = getBaseItemDtoForID(id) ?: return@launch
+			val lifted = baseItem.liftToNewFormat()
 
-			fragment = when (baseItem.baseItemType!!) {
-				BaseItemType.Episode -> {
-					val episode = baseItem.asEpisode()
-					EpisodeDetailsFragment(episode)
+			lifted.acceptVisitor(object : IBaseItemVisitor {
+				override fun visit(item: Episode) {
+					fragment = EpisodeDetailsFragment(item)
 				}
-				else -> TODO()
-			}
+			})
```

Comment:
I would prefer it like this, or is there a reason the visitor pattern would be better in this case?
```kotlin
fragment = when(lifted) {
	is Episode -> EpisodeDetailsFragment(lifted)
	else -> TODO()
}
```

Reply:
It forces you to notice where you missed some changes when a new Item type is introduced, as the interface changes and you have to adjust it everywhere.

Reply:
Did some quick searching and it seems like using sealed classes is a way to do it "the kotlin way". It would look basically the same as my example except for the fact that there is no "else" block.

- https://kotlinlang.org/docs/reference/sealed-classes.html
- https://stackoverflow.com/questions/33602705/best-way-to-implement-visitor-pattern-in-kotlin